File: src/main/java/com/projeto/springapi/model/Endereco.java
package com.projeto.springapi.model;

import jakarta.persistence.*;
import lombok.Data;

@Entity
@Table(name = "endereco")
@Data
public class Endereco {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "end_id")
    private Long endId;

    @Column(name = "end_tipo_logradouro")
    private String endTipoLogradouro;

    @Column(name = "end_logradouro")
    private String endLogradouro;

    @Column(name = "end_numero")
    private Integer endNumero;

    @Column(name = "end_bairro")
    private String endBairro;

    @ManyToOne
    @JoinColumn(name = "cid_id")
    private Cidade cidade;

}

File: app/Models/Endereco.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Model;

class Endereco extends Model
{
    protected $table = 'endereco';
    protected $primaryKey = 'end_id';
    public $timestamps = false;

    protected $fillable = [
        'end_tipo_logradouro',
        'end_logradouro',
        'end_numero',
        'end_bairro',
        'cid_id'
    ];

    public function cidade()
    {
        return $this->belongsTo(Cidade::class, 'cid_id', 'cid_id');
    }
}
